vars/goDefaultVersion.groovy

```diff
@@ -25,9 +25,9 @@ def call(Map args = [:]) {
   if(isGoVersionEnvVarSet()) {
     goDefaultVersion = "${env.GO_VERSION}"
   } else {
-    def goFileVersion = '.go-version'
-    if (fileExists(goFileVersion)){
-      goDefaultVersion = readFile(file: goFileVersion)?.trim()
+    def found = ['.go-version', "${env.BASE_DIR}/.go-version'"].find { fileExists(it) }
```

Comment:
Might be wrong guess, but I found a bug last time in the Windows ecosystem due to a wrong path separator: `\` vs `/`.

Reply:
IIRC, this should be platform agnostic, but let me double check it

Reply:
https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/fcd86dc54c549e1bcc48d13d4ee2ca7e66006253/src/main/java/org/jenkinsci/plugins/workflow/steps/FileExistsStep.java#L85 uses https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/FilePath.java

so it should be platform agnostic then